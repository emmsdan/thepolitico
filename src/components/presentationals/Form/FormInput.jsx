/* eslint-disable no-console */
import React from 'react';
import PropTypes from 'prop-types';

const FormInput = props => {
  const { id, className, title, name, placeholder, defaultValue } = props;
  return (
    <div className="field">
      <input
        {...props}
        className={className}
        name={name}
        id={id || name}
        defaultValue={defaultValue}
        placeholder={placeholder || title}
      />
      <label htmlFor={id || name}>{title}:</label>
    </div>
  );
};

FormInput.propTypes = {
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  title: PropTypes.string.isRequired,
  id: PropTypes.string,
  defaultValue: PropTypes.string,
  placeholder: PropTypes.string,
};

export default FormInput;
