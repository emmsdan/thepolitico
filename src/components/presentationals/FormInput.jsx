/* eslint-disable no-console */
import React from 'react';
import PropTypes from 'prop-types';

const FormInput = props => {
  return (
    <div className="field">
      <input
        className={props.input.class}
        name={props.input.name}
        id={props.input.id}
        value={props.input.value}
        placeholder={props.input.placeholder || ''}
      />
      <label htmlFor={props.input.id}>{props.input.title}:</label>
    </div>
  );
};

FormInput.propTypes = {
  input: PropTypes.shape({
    name: PropTypes.string,
    class: PropTypes.string,
    id: PropTypes.string,
    title: PropTypes.string,
    value: PropTypes.string,
    placeholder: PropTypes.string,
  }),
};

export default FormInput;
