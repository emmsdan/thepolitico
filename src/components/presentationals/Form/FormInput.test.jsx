import React from 'react';
import { shallow } from 'enzyme';

import FormInput from './FormInput';

describe('<FormInput />', () => {
  const input = {
    type: 'submit',
    className: 'bg-gray',
    id: 'emailAddress',
    title: 'Email Address',
    name: 'email',
    placeholder: 'enter your email address',
    defaultValue: 'sample@email.com',
  };

  it('should render correctly', () => {
    const component = shallow(<FormInput {...input} />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should display input label correctly', () => {
    const component = shallow(<FormInput {...input} />);
    expect(component.find('label').text()).toEqual(`${input.title}:`);
    component.unmount();
  });

  it('should display input name correctly', () => {
    const component = shallow(<FormInput {...input} />);
    const inputField = component.find('input');
    expect(inputField.prop('defaultValue')).toEqual(input.defaultValue);
    component.unmount();
  });
});
