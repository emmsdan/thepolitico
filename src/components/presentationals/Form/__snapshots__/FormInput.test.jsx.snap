// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> should render correctly 1`] = `
<div
  className="field"
>
  <span
    className="error red-text"
    htmlFor="emailAddress"
  />
  <input
    className="bg-gray"
    defaultValue="sample@email.com"
    id="emailAddress"
    name="email"
    placeholder="enter your email address"
    title="Email Address"
    type="submit"
  />
  <label
    htmlFor="emailAddress"
  >
    Email Address
    :
  </label>
</div>
`;
